### ISSUE TITLE
Show aggregate function (`estimator`) in axis label

### ISSUE THREAD

**Issue by Rabeez** [mwaskom/seaborn#2370]
I read this in an [article ](https://statmodeling.stat.columbia.edu/2020/10/23/interactive-analysis-needs-theories-of-inference/)about statistical visualization tools:
> ... interactive systems like Tableau default to aggregating data to make high level patterns more obvious but this diminishes some people’s acknowledgment of variation. 

This made me remember how confused I was by the `estimator` argument in seaborn as someone coming from matplotlib. I hadn't yet realized the difference between both libraries and their place in my toolkit but in retrospect (and in light of the quote from the article) showing the effect of this parameter on the plot labels would be much more obvious. This would also be useful otherwise since atleast I always change the axis label anyway if I'm using an aggregation to be clear for the reader. 

I'm curious why the seaborn plots aren't more transparent about the aggregation happening behind the scenes? 

For clarity, the code and output below is basically what I have in mind. 
```python
fig, axes = plt.subplots(ncols=2, nrows=2, constrained_layout=True)
axes = axes.flatten()

sns.lineplot(data=iris, x='sepal_width', y='sepal_length', estimator=None, ax=axes[0])
sns.lineplot(data=iris, x='sepal_width', y='sepal_length', estimator='sum', ax=axes[1])
def my_median(nums):
    return np.median(nums)
sns.barplot(data=iris, x='species', y='sepal_length', estimator=my_median, ax=axes[2])
sns.barplot(data=iris, x='species', y='sepal_length', estimator=lambda d: sum(d), ax=axes[3])

plt.show()
```
![Figure_1](https://user-images.githubusercontent.com/9392057/101078699-545c5280-35c8-11eb-9dd6-a56726d21494.png)

**Comment by Rabeez** [comment]
For the record I'm not advocating for/against implementing this functionality in seaborn. I'm curious about the API design decision and how it affects how (new) users of the library will interpret the graphs.

**Comment by toobamukhtar** [comment]
Thank you for raising this issue! I have been thinking about the same.

**Comment by mwaskom** [comment]
Interesting question.

These functions actually originally did behave this way. But there were some practical issues that caused it to be a source of inconsistency. For example, it didn't work well in a FacetGrid context (the FacetGrid controls the axis labels but doesn't know about the aggregation), and if you are, say, combining a barplot with a stripplot, the y axis label depends on the order the plots are drawn in.

That latter example also demonstrates a theoretical issue ... it's a great idea to represent both the central tendency and individual observations on the same plot, so there's some inconsistency with saying they "should" have different labels. So I remain ambivalent about it but probably think it's best to consider variable assignment and aggregation to be different concepts.

FWIW I read the quoted comment as being about whether the best default kind of plot to draw is one that aggregates vs. one that represent the distribution vs. one that shows the individual datapoints by default. I think I'm broadly sympathetic to this view and it's part of why `catplot` shows a strip plot by default (also discussed in some detail in the [tutorial](http://seaborn.pydata.org/tutorial/categorical.html)). But the question of what sort of plot to show by default is a bit different than what kind of labels to use by default, and once you are asking for a bar plot, I don't really see any way not to do some aggregation (if you have more than one observation for each categorical level).

BTW I recently had the matplotlib folks create a channel on their [discourse board](https://discourse.matplotlib.org/c/3rdparty/seaborn/21) for seaborn and would like to push this kind of discussion over to that forum, which seems better suited for it than github issues. As there's no actionable issue here, I'll close.

**Comment by mwaskom** [comment]
PS I composed most of that answer with categorical plots in mind because the relevant historical context is there but I forgot that you had also mentioned line plots, where there is a question of whether aggregation should happen by default that is separate from the labeling question.

That's a case where different philosophies could push you in different directions but I would tend to say that the non-aggregated repeated measures plot looks weird and is almost never what most people actually want to be make. So the default relational plot is a scatterplot (which doesn't aggregate), but, once you ask for a line plot, aggregation becomes the default, following the seaborn principle that the default behavior should try to be the most useful (at the cost of sometimes being surprising).

**Comment by Rabeez** [comment]
I didn't even think of overlaid plots but your point makes sense. I realized how much of a mess this would be to implement when I made the example above since now `estimator` would have to be a part of the state of the plotter object in all scenarios (not even considering FacetGrid yet). 

> ... the default behavior should try to be the most useful (at the cost of sometimes being surprising).
I agree with this too since newcomers to the package should (hopefully) realize _something_ is happening when they see error bars on their bars.
